src/ucp/proto/proto_select.c

```diff
@@ -1121,7 +1121,9 @@ ucp_proto_select_short_init(ucp_worker_h worker, ucp_proto_select_t *proto_selec
 
         /* If max_msg_length exceeds SSIZE_MAX, use SSIZE_MAX, since short
            protocol thresholds are signed values */
-        max_short_signed = ucs_min(thresh->max_msg_length, SSIZE_MAX);
+        max_short_signed  = ucs_min(thresh->max_msg_length, SSIZE_MAX);
+        max_short_signed -= (max_short_signed < sizeof(uint64_t)) ?
+                            max_short_signed : sizeof(uint64_t);
```

Comment:
this should be part of eager/short protocol, not here (here is common code for eager/hwtm/put/...)